Android bitwarden app stopped having access to vault

### Vaultwarden Support String

### Your environment (Generated via diagnostics page)
* Vaultwarden version: v1.32.5
* Web-vault version: v2024.6.2c
* OS/Arch: linux/x86_64
* Running within a container: true (Base: Debian)
* Environment settings overridden: false
* Uses a reverse proxy: true
* IP Header check: true (X-Real-IP)
* Internet access: true
* Internet access via a proxy: false
* DNS Check: true
* Browser/Server Time Check: true
* Server/NTP Time Check: true
* Domain Configuration Check: false
* HTTPS Check: false
* Database type: SQLite
* Database version: 3.46.0
* Clients used: 
* Reverse proxy and version: 
* Other relevant information: 

### Config (Generated via diagnostics page)
<details><summary>Show Running Config</summary>

**Environment settings which are overridden:** 


```json
{
  "_duo_akey": null,
  "_enable_duo": true,
  "_enable_email_2fa": false,
  "_enable_smtp": true,
  "_enable_yubico": true,
  "_icon_service_csp": "",
  "_icon_service_url": "",
  "_ip_header_enabled": true,
  "_max_note_size": 10000,
  "_smtp_img_src": "cid:",
  "admin_ratelimit_max_burst": 3,
  "admin_ratelimit_seconds": 300,
  "admin_session_lifetime": 20,
  "admin_token": "***",
  "allowed_iframe_ancestors": "",
  "attachments_folder": "data/attachments",
  "auth_request_purge_schedule": "30 * * * * *",
  "authenticator_disable_time_drift": false,
  "data_folder": "data",
  "database_conn_init": "",
  "database_max_conns": 10,
  "database_timeout": 30,
  "database_url": "***************",
  "db_connection_retries": 15,
  "disable_2fa_remember": false,
  "disable_admin_token": false,
  "disable_icon_download": false,
  "domain": "****://*********",
  "domain_origin": "****://*********",
  "domain_path": "",
  "domain_set": false,
  "duo_context_purge_schedule": "30 * * * * *",
  "duo_host": null,
  "duo_ikey": null,
  "duo_skey": null,
  "duo_use_iframe": false,
  "email_2fa_auto_fallback": false,
  "email_2fa_enforce_on_verified_invite": false,
  "email_attempts_limit": 3,
  "email_change_allowed": true,
  "email_expiration_time": 600,
  "email_token_size": 6,
  "emergency_access_allowed": true,
  "emergency_notification_reminder_schedule": "0 3 * * * *",
  "emergency_request_timeout_schedule": "0 7 * * * *",
  "enable_db_wal": true,
  "enable_websocket": true,
  "enforce_single_org_with_reset_pw_policy": false,
  "event_cleanup_schedule": "0 10 0 * * *",
  "events_days_retain": null,
  "experimental_client_feature_flags": "fido2-vault-credentials",
  "extended_logging": true,
  "helo_name": null,
  "hibp_api_key": null,
  "http_request_block_non_global_ips": true,
  "http_request_block_regex": null,
  "icon_blacklist_non_global_ips": true,
  "icon_blacklist_regex": null,
  "icon_cache_folder": "data/icon_cache",
  "icon_cache_negttl": 259200,
  "icon_cache_ttl": 2592000,
  "icon_download_timeout": 10,
  "icon_redirect_code": 302,
  "icon_service": "internal",
  "incomplete_2fa_schedule": "30 * * * * *",
  "incomplete_2fa_time_limit": 3,
  "increase_note_size_limit": false,
  "invitation_expiration_hours": 120,
  "invitation_org_name": "Vaultwarden",
  "invitations_allowed": false,
  "ip_header": "X-Real-IP",
  "job_poll_interval_ms": 30000,
  "log_file": "data/vw.log",
  "log_level": "warn",
  "log_timestamp_format": "%Y-%m-%d %H:%M:%S.%3f",
  "login_ratelimit_max_burst": 10,
  "login_ratelimit_seconds": 60,
  "org_attachment_limit": null,
  "org_creation_users": "",
  "org_events_enabled": false,
  "org_groups_enabled": false,
  "password_hints_allowed": true,
  "password_iterations": 600000,
  "push_enabled": false,
  "push_identity_uri": "https://identity.bitwarden.com",
  "push_installation_id": "***",
  "push_installation_key": "***",
  "push_relay_uri": "https://push.bitwarden.com",
  "reload_templates": false,
  "require_device_email": false,
  "rsa_key_filename": "data/rsa_key",
  "send_purge_schedule": "0 5 * * * *",
  "sendmail_command": null,
  "sends_allowed": true,
  "sends_folder": "data/sends",
  "show_password_hint": false,
  "signups_allowed": false,
  "signups_domains_whitelist": "",
  "signups_verify": false,
  "signups_verify_resend_limit": 6,
  "signups_verify_resend_time": 3600,
  "smtp_accept_invalid_certs": false,
  "smtp_accept_invalid_hostnames": false,
  "smtp_auth_mechanism": null,
  "smtp_debug": false,
  "smtp_embed_images": true,
  "smtp_explicit_tls": null,
  "smtp_from": "",
  "smtp_from_name": "Vaultwarden",
  "smtp_host": null,
  "smtp_password": null,
  "smtp_port": 587,
  "smtp_security": "starttls",
  "smtp_ssl": null,
  "smtp_timeout": 15,
  "smtp_username": null,
  "templates_folder": "data/templates",
  "tmp_folder": "data/tmp",
  "trash_auto_delete_days": null,
  "trash_purge_schedule": "0 5 0 * * *",
  "use_sendmail": false,
  "use_syslog": false,
  "user_attachment_limit": null,
  "user_send_limit": null,
  "web_vault_enabled": true,
  "web_vault_folder": "web-vault/",
  "yubico_client_id": "'removed'",
  "yubico_secret_key": "***",
  "yubico_server": null
}
```
</details>

### Vaultwarden Build Version

Version 2024.6.2

### Deployment method

Official Container Image

### Custom deployment method

No: Plain docker-compose, caddy, fail2ban

### Reverse Proxy

image: caddy:2

### Host/Server Operating System

Linux

### Operating System Version

Ubuntu 22.04.5 LTS

### Clients

Android

### Client Version

latest bitwarden

### Steps To Reproduce

1. Open Bitwarden
2. Enter passcode
3. or log out and Enter Masterpassword

### Expected Result

Successful login

### Actual Result

We were unable to process your request. Please try again or contact us.

### Logs

```text
I have not managed enabling logging yet. I am sure that I will need get logging enabled to get some helpful insight.
```


### Screenshots or Videos

I can log into the vault from any other OS, web browser and extension.
This problem existed a few months ago and disappeared  (reboot phone)
This time I signed up for bitwarden beta which has no newer release and also tried uninstall and reinstall

### Additional Context

It seems that I am logged in as I can change settings but there is no vault access.

> I have found the issue thanks too two people who provided a `har` file. It's a `reprompt` value in the database which does not equals either `NULL`, `0`, or `1`. In both cases they were `4`. How this could have happened I have no idea.
> 
> Adjusting that to either of the valid values will resolve the issue.
> 
> Now, it could be this is not the case for all other users. To debug it a bit better, people could download the `dev` build from the Actions pages, like here: https://github.com/bitwarden/android/actions/runs/12186085669
> 
> Install that `apk`, run it login as usual and see if it still fails. If that is the case, people should install `adb` and run `adb logcat` while there phone is connected via USB or wireless to allow debug access to the phone.
> 
> Click on the retry button, and there should be a stack trace somewhere. That does show a very non-descriptive message which helped to find it.
> 
> A better howto should be created i think, but at least the basic steps are described here now.



Here is the network result

12-08 13:44:30.360 W/NetworkResultCall(29074): Network Error: https://api.bitwarden.com/accounts/prelogin
12-08 13:44:30.360 W/NetworkResultCall(29074): kotlinx.serialization.MissingFieldException: Fields [kdf, kdfIterations] are required for type with serial name 'com.x8bit.bitwarden.data.auth.datasource.network.model.InternalPreLoginResponseJson', but they were missing at path: $
12-08 13:44:30.360 W/NetworkResultCall(29074): 	at com.x8bit.bitwarden.data.platform.datasource.network.serializer.BaseSurrogateSerializer.deserialize(BaseSurrogateSerializer.kt:41)
12-08 13:44:30.360 W/NetworkResultCall(29074): Caused by: kotlinx.serialization.MissingFieldException: Fields [kdf, kdfIterations] are required for type with serial name 'com.x8bit.bitwarden.data.auth.datasource.network.model.InternalPreLoginResponseJson', but they were missing
12-08 13:44:30.360 W/NetworkResultCall(29074): 	at com.x8bit.bitwarden.data.auth.datasource.network.model.InternalPreLoginResponseJson.<init>(InternalPreLoginResponseJson.kt:11)
12-08 13:44:30.360 W/NetworkResultCall(29074): 	at com.x8bit.bitwarden.data.auth.datasource.network.model.InternalPreLoginResponseJson.<init>(Unknown Source:0)
12-08 13:44:30.360 W/NetworkResultCall(29074): 	at com.x8bit.bitwarden.data.auth.datasource.network.model.InternalPreLoginResponseJson$$serializer.deserialize(InternalPreLoginResponseJson.kt:11)
12-08 13:44:30.360 W/NetworkResultCall(29074): 	at com.x8bit.bitwarden.data.auth.datasource.network.model.InternalPreLoginResponseJson$$serializer.deserialize(InternalPreLoginResponseJson.kt:11)


I can attach more if needed

--- a/src/api/core/ciphers.rs
+++ b/src/api/core/ciphers.rs
@@ -511,7 +511,7 @@ pub async fn update_cipher_from_data(
     cipher.fields = data.fields.map(|f| _clean_cipher_data(f).to_string());
     cipher.data = type_data.to_string();
     cipher.password_history = data.password_history.map(|f| f.to_string());
-    cipher.reprompt = data.reprompt;
+    cipher.reprompt = data.reprompt.filter(|r| *r == RepromptType::None as i32 || *r == RepromptType::Password as i32);
 
     cipher.save(conn).await?;
     cipher.move_to_folder(data.folder_id, &headers.user.uuid, conn).await?;

--- a/src/db/models/cipher.rs
+++ b/src/db/models/cipher.rs
@@ -46,10 +46,9 @@ db_object! {
     }
 }
 
-#[allow(dead_code)]
 pub enum RepromptType {
     None = 0,
-    Password = 1, // not currently used in server
+    Password = 1,
 }
 
 /// Local methods
@@ -296,7 +295,7 @@ impl Cipher {
             "creationDate": format_date(&self.created_at),
             "revisionDate": format_date(&self.updated_at),
             "deletedDate": self.deleted_at.map_or(Value::Null, |d| Value::String(format_date(&d))),
-            "reprompt": self.reprompt.unwrap_or(RepromptType::None as i32),
+            "reprompt": self.reprompt.filter(|r| *r == RepromptType::None as i32 || *r == RepromptType::Password as i32).unwrap_or(RepromptType::None as i32),
             "organizationId": self.organization_uuid,
             "key": self.key,
             "attachments": attachments_json,

--- a/src/db/models/mod.rs
+++ b/src/db/models/mod.rs
@@ -18,7 +18,7 @@ mod user;
 
 pub use self::attachment::Attachment;
 pub use self::auth_request::AuthRequest;
-pub use self::cipher::Cipher;
+pub use self::cipher::{Cipher, RepromptType};
 pub use self::collection::{Collection, CollectionCipher, CollectionUser};
 pub use self::device::{Device, DeviceType};
 pub use self::emergency_access::{EmergencyAccess, EmergencyAccessStatus, EmergencyAccessType};

